### PR TITLE
Chore: add modules params to make AWS modules creation configurable 

### PR DESCRIPTION
**Pull request by eladmosh**
### Issue & Steps to Reproduce / Feature Request
currently when the user use our `k8s-modules//aws` module we automatically create all the modules (vpc, eks, efs ...)
### Solution
Make it configurable and let the user decide which resources he wants to create by himself and provision them to our module and which resources he wants us to create

example:
If you want to provision all the components by yourself, create a TF input variable called `modules_info` (var name is open for chnages):
```json
{
	"vpc": {
		"create": false,
		"id": "vpc-092f26b605570e54e",
		"private_subnets": ["subnet-091d4e565774c76bc", "subnet-095cdaad02d57a8d6", "subnet-01c8446e75547eeef", "subnet-0729b32d54b3216c4", "subnet-01e93749f16d100bf"]
	},
	"efs": {
		"create": false,
		"id": "fs-0ed81547beb39d684"
	},
	"eks": {
		"create": false,
		"cluster_id": "elad-eks8"
	},
	"autoscaler": {
		"create": false
	},
	"csi_driver": {
		"create": false
	}
}
```
I decided to put whole the logic in one object to enable validations, apparently TF only support validations inside the same var only.

then use our module this way:
```terraform
module "my-cluster" {
  source = "github.com/env0/k8s-modules//aws"

  region            = var.region
  cluster_name      = var.cluster_name
  modules_info      = var.modules_info
}
```
**Important**: please notice that there are some dependencies between the different modules, for example, you can't provision an `eks` without a `vpc` because the `vpc_id`  is a required attribute when creating resource of type `aws_eks_cluster`. 
Another obvious example is `autoscaler` without an `eks`.
You can find the complete list of illegal var combinations in the `validations` of the variable `modules_info`.

Screen Shot 2022-02-06 at 17 22 58

### CODE REVIEW

File: aws/main.tf [hunk 1]
```diff
@@ -1,6 +1,17 @@
+data aws_eks_cluster "my_eks" {
+  count = var.modules_info.eks.create ? 0 : 1
+  name = var.modules_info.eks.cluster_id
+}
+
+locals {
+  vpc_id = (var.modules_info.efs.create && var.modules_info.eks.create && var.modules_info.vpc.create) ? module.vpc[0].vpc_id : var.modules_info.vpc.id
```
Comment (RLRabinowitz) [inline aws/main.tf]: Why is `var.modules_info.efs.create && var.modules_info.eks.create && var.modules_info.vpc.create` necessary? Isn't `var.modules_info.vpc.create` enough with the variable validations?

File: aws/main.tf [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+  private_subnets = (var.modules_info.efs.create && var.modules_info.eks.create && var.modules_info.vpc.create) ? module.vpc[0].private_subnets : var.modules_info.vpc.private_subnets
```
Comment (RLRabinowitz) [inline aws/main.tf]: Same question

**Review by RLRabinowitz — APPROVED** [review]
Great job! 
Loved what you did with the validations

File: aws/main.tf (hunk 1 above)
Comment (eladmosh) [reply aws/main.tf]: ye, you are right. 
fixing.

File: aws/main.tf (hunk 2 above)
Comment (eladmosh) [reply aws/main.tf]: ☝️

**Pull request merged**